### PR TITLE
New profile filter switch list

### PR DESCRIPTION
**Pull request by jrouzierinverse**
## Description

New profile filter switch list which allow you to match against multiple switches
## Impacts

The portal profile filtering
## NEWS file entries

New Features
++++++++++++
- New profile filter switch list
## Delete branch after merge

NO

### CODE REVIEW

File: t/unittest/profile/filter/switch_list.t [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+use Test::More tests => 8;
+
+#This test will running last
+use Test::NoWarnings;
+use_ok("pf::profile::filter::switch_list");
+
+my $filter = new_ok ( "pf::profile::filter::switch_list", [profile => 'Test', value => 'switch1,switch2,switch3' ],"Test switch_list filter");
+
+#This is the first test
+ok($filter->match({ 'last_switch' => 'switch1'}),"switch1 matches");
```
Comment (julsemaan) [inline t/unittest/profile/filter/switch_list.t]: balance the spaces on each side
{ 'last_switch' => 'switch1' }

File: lib/pf/profile/filter/switch_list.pm [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    $self->lookup(\%hash);
+}
+
+=head1 AUTHOR
+
+Inverse inc. <info@inverse.ca>
+
+=head1 COPYRIGHT
+
+Copyright (C) 2005-2014 Inverse inc.
```
Comment (julsemaan) [inline lib/pf/profile/filter/switch_list.pm]: Wrong year

**Comment by jrouzierinverse** [comment]
This is not ready

**Comment by jrouzierinverse** [comment]
Please review

File: lib/pf/factory/condition/profile.pm [hunk 3]
```diff
@@ -59,6 +59,18 @@ sub instantiate {
             my $cond_and = pf::condition::all->new(conditions => [$cond1, $cond2]);
             return $cond_and;
         }
+        elsif($type eq 'exists_in'){
+            my @values = split(/ *; */, $data->{value});
```
Comment (julsemaan) [inline lib/pf/factory/condition/profile.pm]: whitespace cleanup ?

Comment (julsemaan) [inline lib/pf/factory/condition/profile.pm]: My bad I saw it's done.
But shouldn't we use \s like in the other parts of the code ?

Comment (jrouzierinverse) [inline lib/pf/factory/condition/profile.pm]: yes agreed it is easier to read as shown right now

File: lib/pf/condition/exists_in.pm [hunk 4]
```diff
@@ -0,0 +1,74 @@
+package pf::condition::exists_in;
+
+=head1 NAME
+
+pf::condition::exists_in - is the value exists in a hash
```
Comment (julsemaan) [inline lib/pf/condition/exists_in.pm]: Not sure this is grammatically correct.
Shouldn't it be : 'checks if a value exists in a hash'

File: lib/pf/condition/exists_in.pm [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+has lookup => (
+    is => 'ro',
+    isa => HashRef,
+    required => 1,
+);
+
+=head2 match
+
+Matches if the argument does not match the value
```
Comment (julsemaan) [inline lib/pf/condition/exists_in.pm]: Hum, not sure this is really true about whats happening below

**Comment by jrouzierinverse** [comment]
Please re-review

**Pull request merged**